Runtime/Tracking/Modification/TransformPropertyApplier.cs

```diff
@@ -335,18 +340,35 @@ protected virtual IEnumerator TransitionTransform(TransformData source, Transfor
             WaitForEndOfFrame delayInstruction = new WaitForEndOfFrame();
             while (elapsedTime < transitionDuration)
             {
-                float lerpFrame = (elapsedTime / transitionDuration);
+                float lerpFrame = elapsedTime / transitionDuration;
                 target.transform.SetGlobalScale(Vector3.Lerp(startScale, destinationScale, lerpFrame));
                 target.transform.position = Vector3.Lerp(startPosition, destinationPosition, lerpFrame);
                 target.transform.rotation = Quaternion.Lerp(startRotation, destinationRotation, lerpFrame);
-                elapsedTime += Time.deltaTime;
+                elapsedTime += StartEqualsDestination(startPosition, destinationPosition, startRotation, destinationRotation, startScale, destinationScale) ? transitionDuration : Time.deltaTime;
                 yield return delayInstruction;
             }
 
             target.transform.SetGlobalScale(destinationScale);
-            target.transform.rotation = destinationRotation;
             target.transform.position = destinationPosition;
+            target.transform.rotation = destinationRotation;
             AfterTransformUpdated?.Invoke(eventData.Set(source, target));
         }
+
+        /// <summary>
+        /// Determines whether the start properties equal the destination properties.
+        /// </summary>
+        /// <param name="startPosition">The initial position of the <see cref="Transform"/>.</param>
+        /// <param name="destinationPosition">The final position for the <see cref="Transform"/>.</param>
+        /// <param name="startRotation">The initial rotation of the <see cref="Transform"/>.</param>
+        /// <param name="destinationRotation">The final rotation of the <see cref="Transform"/>.</param>
+        /// <param name="startScale">The initial scale of the <see cref="Transform"/>.</param>
+        /// <param name="destinationScale">The final scale of the <see cref="Transform"/>.</param>
+        /// <returns>Whether the start propeties equal the destination properties.</returns>
+        protected virtual bool StartEqualsDestination(Vector3 startPosition, Vector3 destinationPosition, Quaternion startRotation, Quaternion destinationRotation, Vector3 startScale, Vector3 destinationScale)

```

Comment:
```suggestion
        protected virtual bool ArePropertiesEqual(Vector3 startPosition, Vector3 destinationPosition, Quaternion startRotation, Quaternion destinationRotation, Vector3 startScale, Vector3 destinationScale)
```
In C# the convention seems to be offering special advice for `bool` returning members, including methods: They should answer a question.